assets/js/components/surveys/SurveyViewTrigger.js

```diff
@@ -19,41 +19,47 @@
 /**
  * External dependencies
  */
-import { useMount } from 'react-use';
 import PropTypes from 'prop-types';
 
+/**
+ * WordPress dependencies
+ */
+import { useEffect } from '@wordpress/element';
+
 /**
  * Internal dependencies
  */
 import Data from 'googlesitekit-data';
 import { CORE_SITE } from '../../googlesitekit/datastore/site/constants';
 import { CORE_USER } from '../../googlesitekit/datastore/user/constants';
-
 const { useSelect, useDispatch } = Data;
 
-const SurveyViewTrigger = ( { triggerID, ttl } ) => {
+export default function SurveyViewTrigger( { triggerID, ttl = 0 } ) {
 	const usingProxy = useSelect( ( select ) =>
 		select( CORE_SITE ).isUsingProxy()
 	);
 
+	const isTimedOut = useSelect( ( select ) =>
+		select( CORE_USER ).isSurveyTimedOut( triggerID )
+	);
+	const isTimingOut = useSelect( ( select ) =>
+		select( CORE_USER ).isTimingOutSurvey( triggerID )
+	);
+
 	const { triggerSurvey } = useDispatch( CORE_USER );
 
-	useMount( () => {
-		if ( usingProxy ) {
+	const shouldTriggerSurvey = isTimedOut === false && isTimingOut === false;
+
+	useEffect( () => {
+		if ( shouldTriggerSurvey && usingProxy ) {
 			triggerSurvey( triggerID, { ttl } );
 		}
-	} );
+	}, [ shouldTriggerSurvey, usingProxy, triggerSurvey, triggerID, ttl ] );
```

Comment:
Why is all this needed? It looks like the `triggerSurvey` action already includes these checks and also ensures the API-based bits are resolved. Why duplicate these conditions in here?

I'm not sure why it's needed, and it makes me slightly worried relying on `useEffect` instead of `useMount`, since this really should only happen once.

Reply:
@tofumatt I'll leave it to you and @eugene-manuilov to make a decision on this, but my thinking is that the additional code here is somewhat of a duplicate since the action is already resilient enough to "make sure it works", plus the point about `useEffect` theoretically being more error-prone to calling the action multiple times - while I agree in that I cannot see a point where this happens using the current logic, it feels more error-prone than using `useMount`.

If you still feel the current approach is more resilient, we can leave it, but if you agree, please open a follow-up PR to address this.